packages/core/options.ts

```diff
@@ -102,31 +102,88 @@ export const fromSnakeCaseWithPrefix: (prefix: string) => KeyFormatSource = (
 	};
 };
 
+// Parameters can be passed in via environment variables, command line or
+// configuration file, and subsequently overwrite the default ones and each other.
+// The passed in values have to be set for externally provided parameters, e.g.
+// CLI parameters, before resolving the final options object.
+// Higher index means higher priority.
+export enum ParameterResolverIndex {
```

Comment:
The resolvers could have an explicit "priority". Then we wouldn't need to use enums as array indices/accessors.
If each option knew its origin, we wouldn't need to sort, and could compare the priorities of options instead.

Reply:
I tried a few different designs, but none seemed especially good. 

One problem is that parameters should be passed in and merged with existing ones (default, env vars). This means that their "type" needs to be specifiable externally. Exporting all `Resolvers` could be a way, but that would also reveal too many internal details. Hence, a `Resolver` can now be specified via its index. 

Another option would be to make this completely dynamic and pass whole resolvers or even a list of them into the `buildOptions` function. This feels to open, though.

Do you have another idea how to solve this?

Reply:
I am looking into this, but feel free to merge already, since there are no functional issues!